lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -112,8 +114,9 @@ def preprocess(self, params: ApiParams) -> CONTEXT:
         using the Dataset ownership/access, so validation and authorization has
         already taken place.
         """
+
         _, dataset = self.schemas.get_param_by_type(
-            API_METHOD.POST, ParamType.DATASET, params
+            self.schema.method, ParamType.DATASET, params
```

Comment:
Or `self.schema.get_param_by_type(ParamType.DATASET, params)` since we *know* which method/schema we want to check.